test/e2e/playwright/swap/support/network-controller-page.ts

```diff
@@ -0,0 +1,75 @@
+import { type Locator, type Page } from '@playwright/test';
+
+export class NetworkController {
+  readonly page: Page;
+
+  readonly networkDisplay: Locator;
+
+  readonly addNetworkButton: Locator;
+
+  readonly addNetworkManuallyButton: Locator;
+
+  readonly networkTickerInput: Locator;
+
+  readonly addBtn: Locator;
+
+  readonly approveBtn: Locator;
+
+  readonly saveBtn: Locator;
+
+  readonly switchToNetworkBtn: Locator;
+
+  readonly gotItBtn: Locator;
+
+  readonly networkSearch: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.networkDisplay = this.page.getByTestId('network-display');
+    this.addNetworkButton = this.page.getByText('Add network');
+    this.addNetworkManuallyButton = this.page.getByTestId(
+      'add-network-manually',
+    );
+    this.networkTickerInput = this.page.getByTestId(
+      'network-form-ticker-input',
+    );
+    this.saveBtn = this.page.getByRole('button', { name: 'Save' });
+    this.addBtn = this.page.getByRole('button', { name: 'Add' }).first();
+    this.approveBtn = this.page.getByTestId('confirmation-submit-button');
+    this.switchToNetworkBtn = this.page.locator('button', {
+      hasText: 'Switch to',
+    });
+    this.gotItBtn = this.page.getByRole('button', { name: 'Got it' });
+    this.networkSearch = this.page.locator('input[type="search"]');
+  }
+
+  async addCustomNetwork(option) {
+    await this.networkDisplay.click();
+    await this.addNetworkButton.click();
+    await this.addNetworkManuallyButton.click();
+    const formField = await this.page.$$('.form-field__input');
```

Comment:
is there not a better PW-ish option for this?

Reply:
@racitores I don't understand your comment. Can you please explain?

Reply:
I am talking about this https://playwright.dev/docs/best-practices#use-locators

Reply:
Because of how the code is written there is no better way as far as I know. This is what the code looks like. If you have a better suggestion please let me know

<input class="form-field__input" type="text" value="">
<input class="form-field__input" type="text" value="">
<input class="form-field__input" type="text" value="">
<input class="form-field__input" type="text" value="">
<input class="form-field__input" type="text" value="">